spec/lib/jobs/copy-publickey-spec.js

```diff
@@ -0,0 +1,73 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+var uuid = require('node-uuid');
+
+describe('copy-key job', function() {
+    var waterline = { nodes: {}, catalogs: {} },
+        CopyKeyJob,
+        copyKeyJob,
+        commandUtil = {};
+
+    function CommandUtil() { return commandUtil; }
+
+    before(function() {
+        helper.setupInjector([
+            helper.require('/lib/jobs/copy-publickey.js'),
+            helper.require('/lib/jobs/base-job.js'),
+            helper.di.simpleWrapper(CommandUtil, 'JobUtils.Commands'),
+            helper.di.simpleWrapper({Client:function(){}}, 'ssh'),
+            helper.di.simpleWrapper(waterline, 'Services.Waterline')
+        ]);
+        this.sandbox = sinon.sandbox.create();
+        CopyKeyJob = helper.injector.get('Job.CopyKey');
+    });
+
+    describe('_run', function() {
+        var sshSettings;
+
+        beforeEach(function() {
+            copyKeyJob = new CopyKeyJob({}, {target: 'nodeId'}, uuid.v4());
+            commandUtil.sshExec = this.sandbox.stub().resolves({ stdout: 'success'});
+            sshSettings = {
+                host: 'the remote host',
+                port: 22,
+                username: 'someUsername',
+                password: 'somePassword',
+                publicKey: 'a somewhat long string',
+                privateKey: 'another somewhat long string',
+            };
+            waterline.nodes.needByIdentifier = this.sandbox.stub().resolves(
+                {sshSettings: sshSettings}
+            );
+        });
+
+        it('should use commandUtil.sshExec to copy a key to a remote node', function() {
+            return copyKeyJob._run()
+            .then(function() {
+                expect(commandUtil.sshExec).to.be.calledTwice;
+                expect(commandUtil.sshExec).to.be.calledWithExactly(
+                        {cmd: 'mkdir -p .ssh'},
+                        sshSettings,
+                        {}
+                );
+                expect(commandUtil.sshExec).to.be.calledWithExactly(
+                    {cmd: 'echo '+sshSettings.publicKey+' >> .ssh/authorized_keys'},
+                    sshSettings,
+                    {}
+                );
+            });
+        });
+
+        it('should fail if sshExec fails', function() {
```

Comment:
not strictly needed, but these would benefit from explicit validation of failure in the event of a missing `nodeId` attribute on the workflow and a missing `sshSettings.publicKey` for the `copyKeyJob`.